espnet2/s2st/synthesizer/discrete_synthesizer.py

```diff
@@ -83,17 +82,16 @@ def __init__(
                 assume that spembs will be provided as the input.
             spk_embed_integration_type (str): How to integrate speaker embedding.
         """
-        assert check_argument_types()
         super().__init__()
 
         self.spks = None
         if spks is not None and spks > 1:
             self.spks = spks
-            self.sid_emb = torch.nn.Embedding(spks, encoder_output_size)
+            self.sid_emb = torch.nn.Embedding(spks, idim)
```

Comment:
is it OK?
@ftshijt, can you confirm it.

Reply:
There is no code line initially assigning `encoder_output_size`, So I inferred the corresponding variable based on the TransformerDecoder output size

Reply:
Confirmed, this change should be fine! Thanks!